engines/stark/console.cpp

```diff
@@ -481,12 +557,57 @@ bool Console::Cmd_ChangeLocation(int argc, const char **argv) {
 	uint levelIndex = strtol(argv[1] , nullptr, 16);
 	uint locationIndex = strtol(argv[2] , nullptr, 16);
 
-	StarkResourceProvider->requestLocationChange(levelIndex, locationIndex);
+	bool isIndicesValid = true;
+
+	// Temporarily replace the global archive loader
+	ArchiveLoader *archiveLoader = new ArchiveLoader();
+	ArchiveLoader *gameArchiveLoader = StarkArchiveLoader;
+	StarkArchiveLoader = archiveLoader;
+
```

Comment:
I spent some time deciding how to asset these two indices. One way is to assert them in the actual `requestLocationChange()`, but the codes there seems to be hard to revert when indices are invalid. So I decided to add an additional load and check here. This will load the same resources twice if indices are correct, but codes are easier to read and since the debug console is an extra feature, it should not be an overhead.

Reply:
This seems too complicated to me for the value it brings. IMO, there should be something like `StarkArchiveLoader->canLoadArchive(levelIndex, locationIndex)` that simply checks if the location file exists.